backend/penndata/management/commands/get_rodin_events.py

```diff
@@ -0,0 +1,87 @@
+import datetime
+
+import requests
+from bs4 import BeautifulSoup
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from penndata.models import Event
+
+
+RODIN_EVENTS_WEBSITE = "https://rodin.house.upenn.edu"
+HEADERS = {
+    "User-Agent": "Mozilla/5.0 AppleWebKit/537.36 Chrome/91.0.4472.124 Safari/537.36"
+}
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **kwargs):
+        self.scrape_calendar_page(RODIN_EVENTS_WEBSITE + "/calendar")
+        now = timezone.localtime()
+        current_day, current_month, current_year = now.day, now.month, now.year
+        if current_day > 25:
+            next_month_year = current_year if current_month < 12 else current_year + 1
+            next_month = current_month + 1 if current_month < 12 else 1
+            next_month_url = f"{RODIN_EVENTS_WEBSITE}/calendar/{next_month_year}-{next_month:02d}"
+            self.scrape_calendar_page(next_month_url)
+
+        self.stdout.write("Uploaded Rodin College House Events!")
+
+    def scrape_details(self, event_url):
+        try:
+            resp = requests.get(event_url, headers=HEADERS)
+        except ConnectionError:
+            return None
+        soup = BeautifulSoup(resp.text, "html.parser")
+
+        location = soup.select_one(".field-name-field-room").text.strip() if soup.select_one(
+            ".field-name-field-room") else ""
+        start_time_str = soup.select_one(".date-display-start").get("content") if soup.select_one(
+            ".date-display-start") else ""
+        end_time_str = soup.select_one(
+            ".date-display-end").get("content") if soup.select_one(".date-display-end") else ""
+        start_time = datetime.datetime.strptime(start_time_str, "%Y-%m-%dT%H:%M:%S%z"
+                                                ) if start_time_str else None
+        end_time = datetime.datetime.strptime(end_time_str, "%Y-%m-%dT%H:%M:%S%z"
+                                              ) if end_time_str else None
+        description = soup.select_one(
+            ".field-name-body").text.strip() if soup.select_one(".field-name-body") else ""
+        return location, start_time, end_time, description
+
+    def scrape_calendar_page(self, calendar_url):
+        try:
+            resp = requests.get(calendar_url, headers=HEADERS)
+        except ConnectionError:
+            return None
+        soup = BeautifulSoup(resp.text, "html.parser")
+
+        event_cells = soup.find_all("td", class_="single-day future")
+
+        for cell in event_cells:
+            item = cell.find("div", class_="item")
+            if item:
```

Comment:
nit: personally I'm not a big fan of a lot of nesting because it ends up being hard to read the more layers you get. I would suggest doing something like:
```python3
if not item:
   continue
event_link = item.find("a", href=True)
if not event_link:
   continue
```
That way we can avoid nesting

Reply:
could also use the walrus
`if (item := cell.find("div", class_="item")) and (event_link := item.find("a", href=True))`

Reply:
true!